### ISSUE TITLE
Bug: Desktop Boundaries Innacurate after Sleep

### ISSUE THREAD

**Issue by jedinjapan** [valinet/ExplorerPatcher#2086]
Build 10.0.22621.2215

When resuming use of the system after displays have been put to sleep (but system is on) on a dual monitor system, the active desktop boundaries are incorrectly defined. Aka, when pressing maximize on any window border, the window will occupy only 1/4 of the screen across the top, but use the full width.

Primary display is 4k, Secondary is 1080p.

I suspect for some reason, upon resume the secondary display's dimensions are being used? Issue resolves by restarting Windows Explorer.

**Comment by pyrates999** [comment]
That is a preview update: https://support.microsoft.com/en-us/topic/august-22-2023-kb5029351-os-build-22621-2215-preview-9af25662-083a-43f5-b3a7-975fe25cc692

Uninstall it if you want to use explorer patcher and don't install preview updates in the future: https://github.com/valinet/ExplorerPatcher/wiki/Symbols

In the settings app, untick "Get the latest updates as soon as available" so that preview updates are not automatically installed.

All preview updates are then shown as download and install and are never automatically installed if you have the above unticked.

Preview updates are also not fully tested by Microsoft and thus could have bugs.

Preview updates are not supported by EP.

**Comment by jedinjapan** [comment]
That tracks and I believe you are correct. I was not on the insider releases and that toggle was not set, but was not aware the preview updates would be delivered this way.

I will uninstall later when I can and update to confirm the resolution of the issue.

**Comment by pyrates999** [comment]
To ensure this is a bug with EP, uninstall EP, reboot, and then see if this still happens.

**Comment by Aloki** [comment]
With the latest version of EP the bug does not appear.
OS Build 22631.2715

**Comment by jedinjapan** [comment]
This issue seems to be linked to multiple displays. Possibly scaling related?

One detail I did not mention in the initial report is I operate the 1080p display in portrait mode. Not sure if that might cause it to hold resolution data differently or transpose it somewhere.

**Comment by pyrates999** [comment]
This isn't a bug with EP.  It's related to display port deep sleep.  See here: https://github.com/valinet/ExplorerPatcher/issues/1605#issuecomment-1749376776

**Comment by jedinjapan** [comment]
That is all the more perplexing then if *uninstalling ExplorerPatcher* somehow resolves the bug, because as described by myself and others in this thread, the bug simply is not present when ExplorerPatcher is not in use. So is it a DP deep sleep issue that ExplorerPatcher is re-exposing via the hooked Win10 functions?

I will experiment with this again, but I have not been able to reproduce this bug after uninstalling ExplorerPatcher previously.

**Comment by jedinjapan** [comment]
Changing the rotation of the 2nd monitor results in the size of the incorrect boundary changing when the problem resurfaces. Additionally the taskbar not being excluded from the maximize window function in both monitors regardless of monitor orientation.

And addressing the claim this is just a windows bug, this issue not present when explorerpatcher uninstalled, prior or after.

**Comment by vascotenner** [comment]
I also encounter this issue, when using multiple 1080p monitors. I party have a workaround, by using powertoys and defining fancy zones that exclude the area of the task bar. But this is quite cumbersome when having 10+ windows open.

**Comment by pyrates999** [comment]
This should be closed because it cannot be fixed by EP: https://github.com/valinet/ExplorerPatcher/issues/1605#issuecomment-1749376776

**Comment by Aloki** [comment]
No. I reproduced the bug in other ways: disconnected the HDMI cable from a 2K monitor and connected it to a 4K TV; ran an old game in full screen 480p resolution and exited to a desktop in 2K resolution. In both cases the behavior was the same, the desktop height narrowed to 1/3 of the screen height. Again, the bug never occurred without EP.

**Comment by pyrates999** [comment]
When it happens, open up task manager and kill explorer.exe and then run task manager again and run explorer.exe
